test/Microsoft.AspNetCore.SignalR.Client.Tests/HubConnectionProtocolTests.cs

```diff
@@ -42,15 +41,19 @@ public class HubConnectionProtocolTests
         [Fact]
         public async Task ClientSendsHandshakeMessageWhenStartingConnection()
         {
-            var connection = new TestConnection();
-            var hubConnection = new HubConnection(connection, new JsonHubProtocol(), new LoggerFactory());
+            var connection = new TestConnection(autoNegotiate: false);
+            var hubConnection = new HubConnection(() => connection, new JsonHubProtocol(), new LoggerFactory());
             try
             {
-                await hubConnection.StartAsync();
+                // We can't await StartAsync because it depends on the negotiate process!
+                var startTask = hubConnection.StartAsync().OrTimeout();
```

Comment:
Is the standard to put the OrTimeout with the await or is that my imagination?

Reply:
I don't know that we have a standard. I've mostly seen it with the async method itself, but I kinda agree it would be better with the `await`.